integration/common/variable-serializer/src/main/kotlin/VariableSerializer.kt

```diff
@@ -72,9 +73,15 @@ internal fun Pair<String, Any?>.toJsonPathWithValue(
     "$prefix.${this.first}"
   }
 
-  val value = this.second
+  var value = this.second
   return if (value != null && value.isPrimitiveType()) {
 
+    // trim strings to the value length limit if provided
+    if(valueLengthLimit != null && value is String && value.length > valueLengthLimit) {
+      // TODO: logging
```

Comment:
@zambrovski Do you know if it is possible to have a logger in an extension function?
If its not simple we could also live without it

Reply:
doesn't work nicely